contracts/modules/cover/Cover.sol

```diff
@@ -346,43 +438,45 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard, Mu
 
         // check if this request should be skipped, keeping the previous allocation
         if (poolAllocationRequests[i].skip) {
-          coverSegmentAllocations[allocationRequest.coverId][segmentId].push(previousPoolAllocation);
-          totalCoverAmountInNXM += previousPoolAllocation.coverAmountInNXM;
+          coverSegmentAllocations[allocationRequest.coverId][params.segmentId].push(previousPoolAllocation);
+          vars.totalCoverAmountInNXM += previousPoolAllocation.coverAmountInNXM;
           continue;
         }
 
-        vars.previousPremiumInNXM = previousPoolAllocation.premiumInNXM;
-        vars.refund =
-          previousPoolAllocation.premiumInNXM
-          * (allocationRequest.previousExpiration - block.timestamp) // remaining period
-          / (allocationRequest.previousExpiration - allocationRequest.previousStart); // previous period
-
         // get stored allocation id
         allocationRequest.allocationId = previousPoolAllocation.allocationId;
+
+        uint poolAllocationRatio =
+          previousPoolAllocation.coverAmountInNXM
+          * COVER_ALLOCATION_RATIO_DENOMINATOR
+          / vars.previousTotalCoverAmountInNXM;
+
+        vars.previousAllocationAmountInNXMRepriced =
+          poolAllocationRatio * oldSegmentAmountInNXMRepriced
+          / COVER_ALLOCATION_RATIO_DENOMINATOR;
```

Comment:
this is a bit hard to read - there might be a way to simplify it and some more consistent naming will also help